wazimap_ng/points/views.py

```diff
@@ -211,3 +215,44 @@ def list(self, request, profile_id, geography_code):
         except ObjectDoesNotExist as e:
             logger.exception(e)
             raise Http404
+
+
+class LocationListByDistance(generics.ListAPIView):
+    queryset = models.Location.objects.all().annotate(icon=F('category__profilecategory__theme__icon'),
+                                                      theme_id=F('category__profilecategory__theme__id'),
+                                                      theme_name=F('category__profilecategory__theme__name'),
+                                                      color=F('category__profilecategory__theme__color'),
+                                                      profile_category_id=F('category__profilecategory__id'),
+                                                      profile_category_label=F('category__profilecategory__label'),
+                                                      profile_id=F('category__profilecategory__theme__profile_id'))
+    serializer_class = serializers.LocationThemeSerializer
+
+    def list(self, request, profile_id):
+        search_terms = request.GET.getlist('q', [])
+        lat = float(request.GET.get('lat', None))
+        long = float(request.GET.get('long', None))
+        profile = Profile.objects.get(id=profile_id)
```

Comment:
In case of invalid profile ID this code will throw error

Reply:
for future reference : we just tested it. invalid profile id creates a 404 so it's fine